src/client/BuilderClient.ts

```diff
@@ -81,56 +65,90 @@ export class BuilderClient {
-        }
+      upsertResponse = await this.fetch(`/v1/items/${item.id}`, {
+        headers: { 'content-type': 'application/json' },
+        body: JSON.stringify({ item: { ...item, eth_address: this.address } }),
+        method: 'put'
+      })
+      upsertResponseBody =
+        (await upsertResponse.json()) as ServerResponse<RemoteItem>
+    } catch (error) {
+      throw new ClientError(error.message, undefined, null)
+    }
+
+    if (!upsertResponse.ok || !upsertResponseBody.ok) {
+      throw new ClientError(
+        upsertResponseBody.error ?? 'Unknown error',
+        upsertResponse.status,
+        upsertResponseBody.data
       )
+    }
 
-      if (Object.keys(newContent).length > 0) {
-        const formData = new FormData()
-        for (const path in newContent) {
-          formData.append(item.contents[path], newContent[path])
-        }
+    if (Object.keys(newContent).length > 0) {
+      const formData = new FormData()
+      for (const path in newContent) {
+        formData.append(item.contents[path], newContent[path])
+      }
 
-        await this.axios.post(`/v1/items/${item.id}/files`, formData)
+      let uploadResponse: Response
+
+      try {
+        uploadResponse = await this.fetch(`/v1/items/${item.id}/files`, {
+          body: formData as unknown as BodyInit,
+          method: 'post'
+        })
+      } catch (error) {
+        throw new ClientError(error.message, undefined, null)
       }
 
-      return upsertResponse.data.data
-    } catch (error) {
-      if (axios.isAxiosError(error)) {
+      const uploadResponseBody: ServerResponse<unknown> =
+        await uploadResponse.json()
+
+      if (!uploadResponse.ok || !uploadResponseBody.ok) {
         throw new ClientError(
-          error.response?.data.error,
-          error.response?.status,
-          error.response?.data.error.data
+          uploadResponseBody.error ?? 'Unknown error',
+          upsertResponse.status,
+          uploadResponseBody.data
         )
       }
-      throw error
     }
+
+    return upsertResponseBody.data
   }
 
   /**
    * Gets the content size of an already uploaded content file.
    * @param contentIdentifier - The content hash.
    */
   async getContentSize(contentIdentifier: string): Promise<number> {
+    let contentsResponse: Response
     try {
-      const response = await this.axios.head(
-        `/v1/storage/contents/${contentIdentifier}`
+      contentsResponse = await this.fetch(
+        `/v1/storage/contents/${contentIdentifier}`,
+        { method: 'head' }
       )
-      return Number(response.headers['content-length'])
     } catch (error) {
-      if (axios.isAxiosError(error)) {
-        throw new ClientError(
-          error.response?.data.error,
-          error.response?.status,
-          error.response?.data.error.data
-        )
-      } else {
-        throw error
-      }
+      throw new ClientError(
+        error.response?.data.error,
```

Comment:
A bit cumbersome but should we add an `Unknown Error` here too? like the other cases, if response is not present

Reply:
Oh, this is wrong, if the fetch fails, it shouldn't have data nor a status code. Changed it!